cmd/krew/cmd/search_test.go

```diff
@@ -0,0 +1,58 @@
+// Copyright 2019 The Kubernetes Authors.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package cmd
+
+import (
+	"testing"
+
+	"github.com/google/go-cmp/cmp"
+)
+
+func Test_searchByNameAndDesc(t *testing.T) {
+	keyword := "config"
+	testPlugins := []struct {
+		names    []string
+		descs    []string
+		expected []string
+	}{
+		{
+			names: []string{"switch-config", "ctx", "kc"},
+			descs: []string{
+				" Switches between kubeconfig files",
+				"Switch between contexts in your kubeconfig",
+				"Interactive CRUD operations to manage kubeconfig",
+			},
+			expected: []string{"switch-config", "ctx", "kc"},
+		},
+		{
+			names: []string{"ns", "ctx", "kc"},
+			descs: []string{
+				"Switch between Kubernetes namespaces",
+				"Switch between contexts in your kubeconfig",
+				"Interactive CRUD operations to manage kubeconfig",
+			},
+			expected: []string{"ctx", "kc"},
+		},
+	}
+
+	for _, tp := range testPlugins {
+		t.Run(tp.names[0], func(t *testing.T) {
+			result := searchByNameAndDesc(keyword, tp.names, tp.descs)
+			if diff := cmp.Diff(tp.expected, result); diff != "" {
```

Comment:
Do you think fzf or whatever fuzzy search library would change their algorithm and break this test (as the test expects a particular order)?

Reply:
1. If the search library makes a change, they will ensure the compatibility, as fuzzy search is based on matching scores.
2. We specified the lib version in go.mod, so it is safe to manage the version upgrading.